app/src/main/java/com/hifeful/mealmania/presentation/myMeals/MyMealsFeature.kt

```diff
@@ -10,6 +10,52 @@ import io.reactivex.android.schedulers.AndroidSchedulers
 import io.reactivex.schedulers.Schedulers
 import javax.inject.Inject
 
+/**
+ * 💢 Questions
+ * 🟡 Please, answer on them before moving further and after changes would be made
+ * 1️⃣ How the configuration changes are handled with Feature (if so)?
+ * 2️⃣ Can we have MVI using ViewModel or Presenter?
+ * 3️⃣ Could the race conditions issue happen
```

Comment:
Sorry, `Could we have the race condition issue while using MVI when refreshing the state? I don't mean async calls to Repository, but about refreshing the state in Feature / ViewModel / Presenter`

Reply:
1. I'm using the **MVICore** library to implement **MVI** in the application. This library has a **Timecapsule** that can help to save/restore configuration changes. But I didn't implement it. More info -> https://badoo.github.io/MVICore/features/timecapsule/
2. We can wrap our **Feature** in the **ViewModel** or **Presenter** if we really need it.
3. I don't think so, because **MVI** is implemented using **Observer** pattern. All user actions are passed to the **Observable** flow and handled _individually_. And also **State** in **MVI** is _immutable_, so we copy and create new objects, we don't change existing variables in the **State** object.